presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -188,6 +188,15 @@ public static Slice split(@SqlType(StandardTypes.VARCHAR) Slice string, @SqlType
         return toStackRepresentation(result, VARCHAR);
     }
 
+    @ScalarFunction
+    @SqlType("array<varchar>")
+    public static Slice splitRegexp(@SqlType(StandardTypes.VARCHAR) Slice string, @SqlType(StandardTypes.VARCHAR) Slice pattern)
+    {
+        List<String> result = Splitter.onPattern(pattern.toStringUtf8())
```

Comment:
This will compile the pattern for every invocation. See how the other regexp functions take a `Pattern` which avoids this.